backend/app/db.py

```diff
@@ -987,3 +1015,111 @@ async def verify_punishment(
                 raise NotFound
 
             return await self.get_punishment(punishment_id, conn=conn)
+
+    async def insert_group_event(
+        self,
+        group_id: GroupId,
+        event: GroupEventCreate,
+        created_by: UserId,
+        conn: Optional[Pool] = None,
+    ) -> dict[str, int]:
+        async with MaybeAcquire(conn, self.pool) as conn:
+            query = """INSERT INTO group_events(group_id,
+                                                name,
+                                                description,
+                                                start_time,
+                                                end_time,
+                                                created_by,
+                                                created_time)
+                    VALUES ($1, $2, $3, $4, $5, $6, $7)
+                    RETURNING event_id
+                    """
+            try:
+                group_event_id = await conn.fetchval(
+                    query,
+                    group_id,
+                    event.name,
+                    event.description,
+                    event.start_time,
+                    event.end_time,
+                    created_by,
+                    datetime.datetime.utcnow(),
+                )
+            except UniqueViolationError as exc:
+                raise DatabaseIntegrityException(detail=str(exc)) from exc
+            except ForeignKeyViolationError as exc:
+                raise DatabaseIntegrityException(detail=str(exc)) from exc
+
+        return {"id": group_event_id}
+
+    async def group_event_exists_between(
+        self,
+        group_id: GroupId,
+        start_time: datetime.datetime,
+        end_time: datetime.datetime,
+        conn: Optional[Pool] = None,
+    ) -> bool:
+        async with MaybeAcquire(conn, self.pool) as conn:
+            query = """SELECT 1 FROM group_events
+                WHERE group_id = $1 AND start_time < $2 AND end_time > $3
+                """
+            res = await conn.fetchrow(query, group_id, end_time, start_time)
+            return res is not None
+
+    async def get_total_group_events(
+        self,
+        group_id: GroupId,
+        conn: Optional[Pool] = None,
+    ) -> int:
+        async with MaybeAcquire(conn, self.pool) as conn:
+            query = "SELECT COUNT(*) FROM group_events WHERE group_id = $1"
+            return await conn.fetchval(  # type: ignore
```

Comment:
Rouge comment?

Reply:
pylint doesn't know which type it returns. I guess I could've added an assert isinstance, similar to what I've done other places, but it doesn't really matter that much.

Reply:
Ah makes sense 👍